Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/HDInsightNewAPI/ClusterOperationNewAPIImpl.java

```diff
@@ -66,28 +66,45 @@
             final SubscriptionDetail subscription,
             final String clusterId) throws IOException {
         return getClusterConfigurationRequest(subscription, clusterId)
-                .map(clusterConfiguration -> true)
-                .doOnNext(clusterConfiguration -> setRoleType(HDInsightUserRoleType.OWNER))
+                .map(clusterConfiguration -> {
+                    if (clusterConfiguration != null
+                            && clusterConfiguration.getConfigurations() != null
+                            && clusterConfiguration.getConfigurations().getGateway() != null
+                            && clusterConfiguration.getConfigurations().getGateway().getUsername() != null
+                            && clusterConfiguration.getConfigurations().getGateway().getPassword() != null) {
+                        setRoleType(HDInsightUserRoleType.OWNER);
+                        return true;
+                    } else {
+                        final Map<String, String> properties = new HashMap<>();
+                        properties.put("ClusterID", clusterId);
+                        properties.put("StatusCode", "200");
+                        properties.put("ErrorDetails", "Cluster credential is incomplete.");
+                        AppInsightsClient.createByType(AppInsightsClient.EventType.Telemetry, this.getClass().getSimpleName(), null, properties);
+
+                        log().warn("Cluster credential is incomplete even if successfully get cluster credential.");
```

Comment:
Let's gather feedbacks from this error:

```suggestion
                        log().error("Cluster credential is incomplete even if successfully get cluster credential.");
```